src/axom/sidre/core/Group.cpp

```diff
@@ -1373,6 +1379,67 @@ Group* Group::deepCopyGroup(const Group* srcGroup, int allocID)
   return dstGroup;
 }
 
+/*
+ *************************************************************************
+ *
+ * Deep copy another Group into this Group.
+ *
+ * All contents in this group and their data are destroyed before the copy.
+ * The deep copy of a Group will copy the group hierarchy and deep copy
+ * all Views within the hierarchy.
+ *
+ *************************************************************************
+ */
+Group* Group::deepCopyGroupToSelf(const Group* srcGroup)
+{
+  SLIC_ERROR_IF(m_is_list && !srcGroup->m_is_list,
+                "Group::deepCopyToSelf cannot copy from a list Group '" + srcGroup->getPath() +
+                  "' to a non-list Group '" + getPath() + "'");
+
+  destroyGroupsAndData();
+  destroyViewsAndData();
+
+  // copy child Groups to new Group
+  for(auto& grp : srcGroup->groups())
+  {
+    deepCopyGroup(&grp);
+  }
+
+  // copy Views to new Group
+  for(auto& view : srcGroup->views())
+  {
+    deepCopyView(&view);
+  }
+
+  return this;
+}
+
+/*
+ *************************************************************************
+ *
+ * Reallocate data to View-specific allocators.
+ *
+ *************************************************************************
+ */
+Group* Group::reallocateTo(const std::function<int(const View&)>& viewToAllocatorId)
+{
+  for(auto& grp : groups())
+  {
+    grp.reallocateTo(viewToAllocatorId);
+  }
+
+  for(auto& view : views())
+  {
+    int newAllocId = viewToAllocatorId(view);
+    if(newAllocId != axom::INVALID_ALLOCATOR_ID)
+    {
+      view.reallocateTo(newAllocId);
+    }
```

Comment:
Should there be an `else`? Or does the user expect to only reallocate when the allocator is valid?

Reply:
As mentioned above, the `reallocateTo` method will be removed.